File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    let answer = [];
    const row = arr.length;
    const col = arr[0].length;
    
    if (row > col) {
        for (let i = 0; i < row; i += 1) {
            const newArr = [...arr[i]];
            for (let j = col; j < row; j += 1) {
                newArr.push(0);
            }
            answer.push(newArr);
        }
    } else if (row < col) {
        for (let i = 0; i < col; i += 1) {
            if (i < row) {
                answer.push(arr[i]);
            } else {
                const newArr = Array(col).fill(0);
                answer.push(newArr);
            }
        }
    } else {
        answer = arr;
    }
    return answer;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    answer = []
    row = len(arr)
    col = len(arr[0]) if arr else 0

    if row > col:
        for i in range(row):
            new_row = list(arr[i])
            for _ in range(row - col):
                new_row.append(0)
            answer.append(new_row)
    elif row < col:
        for i in range(col):
            if i < row:
                answer.append(list(arr[i]))
            else:
                new_row = [0] * col
                answer.append(new_row)
    else:
        answer = [list(r) for r in arr]
        
    return answer